[13.0][FIX] pos_payment_terminal: Graceful cancel handling

thomaspaulb opened the pull request:
Because `send_payment_cancel` was not implemented and the original Odoo version does not return a promise at all, I got the following error on payment cancel:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'finally') [point_of_sale.assets.js:268]
```

Not only does this error look funny in the Javascript console, but I also got problems with black screen when running POS in debug mode, because apparently once a payment is cancelled but it's not handled with a promise, it retries on page load, causing a Javascript error on page load.

While of course the best course of action is to implement the actual cancel flow, this was out of scope for the moment and so I just display a popup telling the user to cancel the payment with the red button - and then set the promise as resolved.

Commit:
[IMP] pos_payment_terminal: Graceful cancel handling

sbidoul commented:
@thomaspaulb thanks for this. LGTM. Could you look after the pre-commit issue ?

sbidoul commented on pos_payment_terminal/static/src/js/payment_terminal.js:
```diff
@@ -169,11 +177,21 @@ odoo.define("pos_payment_terminal.payment", function(require) {
         },
+
+        _show_tip: function(msg, title) {
+            if (!_.isEmpty(this.pos.gui.popup_instances))
```
pre-commit complains about this funny indentation

sbidoul commented:
I ported this to 14 in #741

sbidoul committed:
[FIX] pos_payment_terminal: reject cancellation

Since we can't cancel the transaction, we return a rejected
promise. The user should cancel manually which will
effectively free the transaction and let the user delete it.

ddejong-therp commented:
LGTM.

I've setup a PoS setup with a faulty payment terminal module that does not return a response back (does not implement get_status properly in the driver).

When I make a payment this faulty driver causes the interface to never complete the transaction, waiting indefinitely. When I then refresh the page, it stays on a black (dark grey) screen after loading, never really entering into the pos screen.

This MR fixes that, allowing you to enter the pos screen again.

ddejong-therp reviewed (approved):
As stated in the comments: LGTM

ivantodorovich commented:
/ocabot merge patch

ivantodorovich commented:
Let's retry merging this after:
- https://github.com/OCA/pos/pull/795

ivantodorovich commented:
/ocabot merge minor
